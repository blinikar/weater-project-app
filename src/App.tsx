import React from "react";
import styles from "App.module.scss";
import { Link, Outlet } from "react-router-dom";

function App() {
  return (
    <>
      <div className={styles.header}>
        <Link to="/">
          <h1 className={styles.heading}>Weather</h1>
        </Link>

        <nav>
          <ul>
            <Link to="/">
              <li>Home</li>
            </Link>
            <Link to="/map">
              <li>Map</li>
            </Link>
          </ul>
        </nav>
      </div>
      <Outlet />
    </>
  );
}

export default App;
