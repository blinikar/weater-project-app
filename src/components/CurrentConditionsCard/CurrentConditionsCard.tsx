import React from "react";
import styles from "components/CurrentConditionsCard/CurrentConditionsCard.module.scss";
import { useDayCardLogic } from "components/CurrentConditionsCard/CurrentConditionsCard.logic";
import { City } from "models/City";

export interface CurrentConditionsCardProps {
    city: City;
    temp: number;
    type: "cloudy" | "sunny" | "rainy" | "thunderstorm";
    date: Date;
}

export const CurrentConditionsCard:React.FunctionComponent<CurrentConditionsCardProps> = (
    props
) => {

    const logic = useDayCardLogic(props);
    const { temp, type, date, city } = props;

    const [imageUrl, backgroundUrl] = logic.useBackgroundImageUrl();
    const [isInStarred, onStarredClick] = logic.useStarredCities();

    return <div
        className={styles["current-card"]}
        style={{
            backgroundImage: `url('${imageUrl}')`,
            color: backgroundUrl
        }}
    >
        <div className={styles["conditions"]}>
            Current conditions:
            <h1>{temp}K</h1>
            <p>{type}</p>
            <p className={styles["last"]}>{`${date.getDay()}.${date.getMonth()}.${date.getFullYear()}`}</p>
        </div>
        <div className={styles["location-info"]}>
            <span>
                <h1>{city.name}</h1>
                <button onClick={onStarredClick}>
                    <img
                        src={isInStarred ? "star_black_24dp.svg" : "star_outline_black_24dp.svg"} alt={"star"}
                    />
                </button>
            </span>
        </div>
    </div>
}
//Â°C