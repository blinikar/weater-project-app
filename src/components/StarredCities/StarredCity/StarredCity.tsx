import React from "react";
import styles from "components/StarredCities/StarredCity/StarredCity.module.scss";
import { useNavigate } from "react-router-dom";

export interface StarredCityProps {
    name: string;
    link: string;
    onRemoveClick: () => void;
}

export const StarredCity:React.FunctionComponent<StarredCityProps> = (props) => {

    const { name, link, onRemoveClick } = props;

    const navigate = useNavigate()

    return <div className={styles["starred-city"]}>
        <button onClick={() => {navigate(link)}}>{ name }</button>
        <button
            onClick={(e) => {e.stopPropagation(); onRemoveClick()}}
            className={styles["remove"]}
        >
            <img src={"close_FILL0_wght400_GRAD0_opsz48.svg"} alt={"remove"} />
        </button>
    </div>
}