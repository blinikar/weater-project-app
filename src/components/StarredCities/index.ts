export * from "components/CurrentConditionsCard/CurrentConditionsCard";
export * from "components/CurrentConditionsCard/CurrentConditionsCard.logic";
