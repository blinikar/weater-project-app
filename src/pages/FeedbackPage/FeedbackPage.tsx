import React from "react";
import styles from "pages/FeedbackPage/FeedbackPage.module.scss";
import {
  FormValues,
  useWelcomePageLogic,
} from "pages/FeedbackPage/FeedbackPage.logic";
import {
  ErrorMessage, Field, Form, Formik,
} from "formik";
import { Link } from "react-router-dom";

export interface FeedbackPageProps {}

export const FeedbackPage: React.FunctionComponent<FeedbackPageProps> = (
  props,
) => {
  const logic = useWelcomePageLogic(props);

  const [initialValues, validator, submit] = logic.useFormik();
  const [isSubmitted, afterSubmit] = logic.useSubmit();

  return (
    <div className={styles["feedback-page"]}>
      {isSubmitted ? (
        <div>
          <h1>Thank you</h1>
          <p>Your opinion is very important for you</p>
          <Link to="/">Return to the forecast</Link>
        </div>
      ) : (
        <>
          <div>
            <h1>Feedback</h1>
            <p>
              Leave your feedback if weather conditions in your city are wrong
            </p>
          </div>

          <Formik
            initialValues={initialValues}
            validate={validator}
            onSubmit={(
              values: FormValues,
              {
                setSubmitting,
              }: {
                setSubmitting: (state: boolean) => void;
              },
            ) => {
              submit(values, { setSubmitting });
              afterSubmit();
            }}
          >
            {({ isSubmitting }) => (
              <Form className={styles.form}>
                <h3>Personal info</h3>
                <div className={styles.field}>
                  <Field type="text" name="name" placeholder="Your Name*" />
                  <ErrorMessage name="name" component="div" />
                </div>
                <div className={styles.field}>
                  <Field type="email" name="email" placeholder="Your Email*" />
                  <ErrorMessage name="email" component="div" />
                </div>
                <div className={styles.field}>
                  <Field type="text" name="city" placeholder="Your City*" />
                  <ErrorMessage name="city" component="div" />
                </div>
                <h3>Current conditions</h3>

                <div className={styles.field}>
                  <Field
                    type="text"
                    name="temperature"
                    placeholder="Current temperature"
                  />
                  <ErrorMessage name="temperature" component="div" />
                </div>
                <div className={styles.field}>
                  <Field
                    type="text"
                    name="weatherType"
                    placeholder="Current weather type"
                  />
                  <ErrorMessage name="weatherType" component="div" />
                </div>
                <div className={styles.field}>
                  <Field
                    type="text"
                    name="conditions"
                    placeholder="Additional info"
                    rows={5}
                    component="textarea"
                  />
                  <ErrorMessage name="conditions" component="div" />
                </div>
                <button type="submit" disabled={isSubmitting}>
                  Submit
                </button>
              </Form>
            )}
          </Formik>
        </>
      )}
    </div>
  );
};
