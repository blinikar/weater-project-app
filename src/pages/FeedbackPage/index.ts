export * from "pages/FeedbackPage/FeedbackPage";
export * from "pages/FeedbackPage/FeedbackPage.logic"