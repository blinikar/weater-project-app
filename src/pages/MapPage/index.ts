export * from "pages/MapPage/MapPage";
export * from "pages/MapPage/MapPage.logic";
