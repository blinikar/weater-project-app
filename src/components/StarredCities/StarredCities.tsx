import React from "react";
import styles from "components/StarredCities/StarredCities.module.scss";
import { useStarredCitiesLogic } from "components/StarredCities/StarredCities.logic";
import { StarredCity } from "components/StarredCities/StarredCity/StarredCity";
import { useAppDispatch, useAppSelector } from "store";
import { starredCitiesActions } from "store/StarredReducer";

export interface StarredCitiesProps {}

export const StarredCities: React.FunctionComponent<StarredCitiesProps> = (
  props,
) => {
  const logic = useStarredCitiesLogic(props);

  const cities = useAppSelector((state) => state.starredCities);
  const dispatch = useAppDispatch();

  return (
    <div className={styles["starred-cities"]}>
      {cities.map((e, i) => (
        <StarredCity
          name={e.name}
          link={e.link}
          key={i}
          onRemoveClick={() => {
            dispatch(starredCitiesActions.toggleCity(e));
          }}
        />
      ))}
    </div>
  );
};
