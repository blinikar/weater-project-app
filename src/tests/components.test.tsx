import renderer from "react-test-renderer";
import { CurrentConditionsCard } from "components/CurrentConditionsCard";
import { City } from "models/City";
import { StarredCity } from "components/StarredCities/StarredCity/StarredCity";
import { BrowserRouter } from "react-router-dom";
import App from "App";
import { Provider } from "react-redux";
import store from "store";
import { StarredCities } from "components/StarredCities/StarredCities";
import { WelcomePage } from "pages/WelcomePage";
import { FeedbackPage } from "pages/FeedbackPage";
import { MapPage } from "pages/MapPage";
import { NoPage } from "pages/NoPage";

it("Starred city renders correctly", () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <StarredCity
          link="/innopolis"
          name="Innopolis"
          onRemoveClick={() => {
            console.log("bla bla");
          }}
        />
      </BrowserRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("Current conditions card renders correctly", () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <CurrentConditionsCard
          type="Clear"
          temp={10}
          city={new City("Innopolis", "/innopolis")}
          date={new Date(2022, 1, 1)}
        />
        )
      </Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("Starred cities renders correctly", () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <Provider store={store}>
          <StarredCities />
        </Provider>
      </BrowserRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("Welcome page renders correctly", () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <Provider store={store}>
          <WelcomePage />
        </Provider>
      </BrowserRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("Feedback page renders correctly", () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <Provider store={store}>
          <FeedbackPage />
        </Provider>
      </BrowserRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("Map page renders correctly", () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <Provider store={store}>
          <MapPage />
        </Provider>
      </BrowserRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("Layout page renders correctly", () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <Provider store={store}>
          <App />
        </Provider>
      </BrowserRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("No page renders correctly", () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <NoPage />
      </BrowserRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
