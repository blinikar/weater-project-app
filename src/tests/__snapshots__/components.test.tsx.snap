// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Current conditions card renders correctly 1`] = `
Array [
  <div
    className="current-card"
    style={
      Object {
        "backgroundImage": "url('https://upload.wikimedia.org/wikipedia/commons/0/07/Clear_Sky.jpg')",
        "color": "black",
      }
    }
  >
    <div
      className="conditions"
    >
      Current conditions:
      <h1>
        -264
        Â°C
      </h1>
      <p>
        Clear
      </p>
      <p
        className="last"
      >
        0:0
      </p>
    </div>
    <div
      className="location-info"
    >
      <span>
        <h1>
          Innopolis
        </h1>
        <button
          onClick={[Function]}
        >
          <img
            alt="star"
            src="star_outline_black_24dp.svg"
            style={
              Object {
                "filter": "",
              }
            }
          />
        </button>
      </span>
    </div>
  </div>,
  ")",
]
`;

exports[`Feedback page renders correctly 1`] = `
<div
  className="feedback-page"
>
  <div>
    <h1>
      Feedback
    </h1>
    <p>
      Leave your feedback if weather conditions in your city are wrong
    </p>
  </div>
  <form
    action="#"
    className="form"
    onReset={[Function]}
    onSubmit={[Function]}
  >
    <h3>
      Personal info
    </h3>
    <div
      className="field"
    >
      <input
        name="name"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="Your Name*"
        type="text"
        value=""
      />
    </div>
    <div
      className="field"
    >
      <input
        name="email"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="Your Email*"
        type="email"
        value=""
      />
    </div>
    <div
      className="field"
    >
      <input
        name="city"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="Your City*"
        type="text"
        value=""
      />
    </div>
    <h3>
      Current conditions
    </h3>
    <div
      className="field"
    >
      <input
        name="temperature"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="Current temperature"
        type="text"
        value=""
      />
    </div>
    <div
      className="field"
    >
      <input
        name="weatherType"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="Current weather type"
        type="text"
      />
    </div>
    <div
      className="field"
    >
      <textarea
        name="conditions"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="Additional info"
        rows={5}
        type="text"
      />
    </div>
    <button
      disabled={false}
      type="submit"
    >
      Submit
    </button>
  </form>
</div>
`;

exports[`Layout page renders correctly 1`] = `
<div
  className="header"
>
  <a
    href="/"
    onClick={[Function]}
  >
    <h1
      className="heading"
    >
      Weather
    </h1>
  </a>
  <nav>
    <ul>
      <a
        href="/"
        onClick={[Function]}
      >
        <li>
          Home
        </li>
      </a>
      <a
        href="/map"
        onClick={[Function]}
      >
        <li>
          Map
        </li>
      </a>
    </ul>
  </nav>
</div>
`;

exports[`Map page renders correctly 1`] = `
<div
  className="map-page"
>
  <div
    className="map"
  />
</div>
`;

exports[`No page renders correctly 1`] = `
<div
  className="no-page"
>
  <h1>
    This page does not exists :(
  </h1>
  <p>
    We have another pages, there are also beautiful ones
  </p>
  <a
    href="/"
    onClick={[Function]}
  >
    To the Forecast
  </a>
</div>
`;

exports[`Starred cities renders correctly 1`] = `
<div
  className="starred-cities"
>
  <div
    className="starred-city"
  >
    <button
      onClick={[Function]}
    >
      Innopolis
    </button>
    <button
      className="remove"
      onClick={[Function]}
    >
      <img
        alt="remove"
        src="close_FILL0_wght400_GRAD0_opsz48.svg"
      />
    </button>
  </div>
</div>
`;

exports[`Starred city renders correctly 1`] = `
<div
  className="starred-city"
>
  <button
    onClick={[Function]}
  >
    Innopolis
  </button>
  <button
    className="remove"
    onClick={[Function]}
  >
    <img
      alt="remove"
      src="close_FILL0_wght400_GRAD0_opsz48.svg"
    />
  </button>
</div>
`;

exports[`Welcome page renders correctly 1`] = `
<div
  className="welcome-page"
>
  <div
    className="starred-cities"
  >
    <div
      className="starred-city"
    >
      <button
        onClick={[Function]}
      >
        Innopolis
      </button>
      <button
        className="remove"
        onClick={[Function]}
      >
        <img
          alt="remove"
          src="close_FILL0_wght400_GRAD0_opsz48.svg"
        />
      </button>
    </div>
  </div>
  <div
    className="skeleton"
  />
  <div
    className="day-cards"
  />
  <a
    href="/feedback"
    onClick={[Function]}
  >
    <p>
      Wrong forecast?
    </p>
  </a>
</div>
`;
